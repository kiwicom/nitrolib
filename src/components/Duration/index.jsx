// @flow strict
import * as React from "react";
import dateFnsFormat from "date-fns/format";
import * as locales from "date-fns/locale";
import addSeconds from "date-fns/addSeconds";
import differenceInSeconds from "date-fns/differenceInSeconds";

import { fixDurationFormat } from "../../records/LangInfo";
import { Consumer } from "../../services/intl/context";

type Props = {
  to: Date,
  from: Date,
  format: string,
};

const Duration = ({ from, to, format }: Props) => {
  const date = addSeconds(new Date(0), differenceInSeconds(to, from));
  return (
    <Consumer>
      {intl =>
        dateFnsFormat(date, format || fixDurationFormat(intl.language.durationFormat), {
          locale: locales[intl.language.locations] || locales.enUS,
        })
      }
    </Consumer>
  );
};

Duration.defaultProps = {
  format: "",
};

export default Duration;
