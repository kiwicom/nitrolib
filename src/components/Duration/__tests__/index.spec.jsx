// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import Duration from "..";

describe("#Duration", () => {
  test("format default", () => {
    const dateFrom = new Date(Date.UTC(2017, 9, 28));
    const dateTo = new Date(Date.UTC(2017, 9, 30));

    const wrapper = shallow(<Duration timeFrom={dateFrom} timeTo={dateTo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("format custom", () => {
    const dateFrom = new Date(Date.UTC(2017, 9, 28));
    const dateTo = new Date(Date.UTC(2017, 9, 30));

    const wrapper = shallow(<Duration timeFrom={dateFrom} timeTo={dateTo} format="HH:mm:ss" />);

    expect(wrapper).toMatchSnapshot();
  });
});
