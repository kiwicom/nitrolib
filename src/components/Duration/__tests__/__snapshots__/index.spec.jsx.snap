// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Duration format custom 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`#Duration format default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
